File: Document/beginners/sample_mruby/sample03/app/tracer.rb
class Tracer

  PWM_MAX = 100
  PWM = PWM_MAX / 6
  THRESHOLD = 20

  # コンストラクタ
  #
  # 引数でポートを渡せる、省略時はデフォルトのポートを使う
  def initialize(left_motor_port=:port_c,
                 right_motor_port=:port_b,
                 color_port=:port_3)
    @left_wheel = EV3RT_TECS::Motor.new(left_motor_port)
    @right_wheel = EV3RT_TECS::Motor.new(right_motor_port)
    @color_sensor = EV3RT_TECS::ColorSensor.new(color_port)
  end

  def init
    Util::Display.init(self.class.to_s)
  end

  def terminate
    Util::Display.message("Stopped", 2)
    @right_wheel.stop
    @left_wheel.stop
  end
  
  def run
    Util::Display.message("running...", 2)
    if @color_sensor.reflect >= THRESHOLD
      @left_wheel.power = 0
      @right_wheel.power = PWM
    else
      @left_wheel.power = PWM
      @right_wheel.power = 0
    end
  end

end

File: Document/beginners/sample_java/sample03/src/jp/etrobo/ev3/sample03/app/Tracer.java
/*
 *  Tracer.java (for leJOS EV3)
 *  Created on: 2016/06/11
 *  Copyright (c) 2016 Embedded Technology Software Design Robot Contest
 */
package jp.etrobo.ev3.sample03.app;

import jp.etrobo.ev3.sample03.ColorSensor;
import lejos.hardware.lcd.LCD;
import lejos.hardware.port.MotorPort;
import lejos.hardware.port.SensorPort;
import lejos.hardware.port.TachoMotorPort;

public class Tracer {
    private TachoMotorPort leftWheel;
    private TachoMotorPort rightWheel;
    private ColorSensor    colorSensor;

	private static final int pwm = TachoMotorPort.MAX_POWER / 6;
    private static final float THRESHOLD = 0.2F;

    public Tracer(){
    	leftWheel = MotorPort.C.open(TachoMotorPort.class);
    	rightWheel = MotorPort.B.open(TachoMotorPort.class);
    	colorSensor = new ColorSensor(SensorPort.S3);
    }

    public void init(){
    	LCD.drawString("Tracer", 0, 2);
    }
    public void run(){
    	LCD.drawString("running...", 0, 4);
        if(colorSensor.getBrightness() >= THRESHOLD){
        	leftWheel.controlMotor(0, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        }else{
        	leftWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(0, TachoMotorPort.PWM_FLOAT);
        }
    }
    public void terminate(){
        LCD.drawString("Stopped.", 0, 4);
        leftWheel.setPWMMode(TachoMotorPort.STOP);
        rightWheel.setPWMMode(TachoMotorPort.STOP);

    }
}
